### PR TITLE
Add 512-bit variation of OFPortBigMap

### PR DESCRIPTION
**Pull request by shudongz**
Reviewer: @rlane 

A dup of original class, changing 128 bit into 512 bit. Add unit tests.

### CODE REVIEW

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com//job/loxigen_pull_req/149/
Test PASSed.

File: java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/OFPortBitMap512.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+         *  will NOT be matched.
+         */
+        public Builder set(OFPort port) {
+            // see the implementation note above about the logical inversion of the mask
+            int bit = port.getPortNumber();
+            if (bit < 0 || bit > 511)
+                throw new IndexOutOfBoundsException("Port number is out of bounds");
+            else if (bit == 511)
+                // the highest order bit in the bitmask is reserved. The switch will
+                // set that bit for all ports >= 127. The reason is that we don't want
```
Comment (rlane) [inline java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/OFPortBitMap512.java]: Comment should be updated for larger size.

**Comment by rlane** [comment]
Minor comment, otherwise looks good.

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com//job/loxigen_pull_req/150/
Test PASSed.

**Comment by rlane** [comment]
ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2015.06.10.0016-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2015.06.10.0016-m.loxigen.master/abat.log
/cc

**Pull request merged**